### PR TITLE
Telstra coding exercise 2 - transform payload

### PR DESCRIPTION
**Pull request by harsingh10**
Changes made:

1. Exposed a POST endpoint which accepts an input JSON object with payload, referenceData and returns a transformedPayload (JSON)
2. Dockerised the app (DockerFile)

### CODE REVIEW

File: utils/transformedPayload.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    const referenceKeys = Object.keys(referenceData);
+
+    // Transofrmation logic   
+
+    // return if key != value
+    if (typeof transformedPayload.value !== 'string') {
+        transformedPayload.value.forEach((valueObj) => { valueObj = transformPayload(valueObj, referenceData) })
+    } else {
+        // check if value of 'value' key is present in referenceKeys
+        referenceKeys.forEach(refKey => {
```
Comment (sam-dev-challenge) [inline utils/transformedPayload.js:13]: your looping through each property in `referenceData` for every string value in `payload`. If the `referenceData` contains 100's of properties but your payload may not necessarily contain all of these properties is it worth looping through all ? Can this be made more efficient ?

File: utils/transformedPayload.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+const transformPayload = (payload, referenceData) => {
```
Comment (sam-dev-challenge) [inline utils/transformedPayload.js:1]: This function mutates the original payload in `req`. If the original payload is used elsewhere mutating it can cause serious issues. Can this be made a pure function ?

Comment (sam-dev-challenge) [inline utils/transformedPayload.js:1]: As your application grows this function might be used in other parts of your code. Think of it like a utility function and can you add unit tests to test this function for positive and negative scenarios.

File: test/data.test.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
+const request = require('supertest');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
+
+const app = require('../app');
+const expect = chai.expect;
+
+chai.use(chaiAsPromised);
+
+describe("functional -data ", () => {
```
Comment (sam-dev-challenge) [inline test/data.test.js:10]: This file is pretty long just for a single test. Can this be organised in a better way ?
